numba/tests/test_slices.py

```diff
@@ -185,6 +185,23 @@ def test_slice_indices_examples(self):
             str(e.exception)
         )
 
+    def test_slice_constant_lowering(self):
+        for ts in [slice(1, 3, None), slice(2, None, None), slice(None, 2, -1),
+                   slice(None, None, None), slice(None, None, -2)]:
+            test_tuple = (1, 2, 3, 4)
+
+            @jit(nopython=True)
+            def test_fn():
+                return test_tuple[ts]
```

Comment:
I think that this test is probably a good test of jit with parallel=True option as well.  Maybe @sklam or @stuartarchibald could comment but it isn't clear to me whether another test_fn_par should be added here with parallel=True (and thus parallel testing is not all in test_parfors) or whether this code should be copied to test_parfors (in which case we have some code duplication).  I lean towards copying it to test_parfors.  Look for other slice tests in there.